tests/kernel/mem_protect/mem_protect/src/kobject.c

```diff
@@ -886,3 +944,120 @@ void test_create_new_invalid_prio_thread_from_user(void *p1, void *p2, void *p3)
 
 	k_thread_join(&child_thread, K_FOREVER);
 }
+
+/* Function to check read/write access to thread's own stack memory buffer */
+static void thread_stack_rw_access(void *p1, void *p2, void *p3)
+{
+	set_fault_valid(false);
+
+	u32_t stack_buf[0];
+	/* write stack data */
+	stack_buf[0] = 1234;
```

Comment:
this doesn't really test that you can read/write the entire stack buffer, just the first 4 bytes.

Given that tests/kernel/mem_protect/userspace already check this, why was this added?
```
		/* If we're in user mode, check every byte in the stack buffer
		 * to ensure that the thread has permissions on it.
		 */
		for (pos = stack_start; pos < stack_end; pos++) {
			zassert_false(check_perms((void *)pos, 1, 1),
				      "bad MPU/MMU permission on stack buffer at address %p",
				      pos);
		}
```

Reply:
@andrewboie As I understand I can remove my test test_stack_user_rw_access, because it is already tested. Right?

Reply:
you should squash the commits, adding a test in 1 commit and then removing it in the next has no value.

Reply:
@nashif Updated.